src/insertion_sort_arr_ll.c

```diff
@@ -0,0 +1,192 @@
+/**
+ *@brief This Module includes all function needed for performing a cpu runtime measurement assessing the performance of the insertionsort algorithm applied on a array and a linked list
+ *
+ *
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <math.h>
+#include <string.h>
+#include "../include/node_t.h"
+#include "../include/randomizer_8_16_64.h"
+#include "../include/randomizer_pt.h"
+#include "../include/insertionsort.h"
+#include "../include/sortcheck.h"
+#include "../include/mergesort.h"
+#include "../include/search_index.h"
+#include "../include/quicksort.h"
+
+/**
+ *@brief This function appends a new node with a given number element
+ *@param[in] head Head of current list
+ *@param[in] num Number element of new node
+ *
+ */
+void append_node(node_t **head, int num){
+
+    node_t *ptr = *head;
+
+    node_t *new = malloc(sizeof(node_t));
+    new->num = num;
+    new->next = NULL;
+
+    if(*head == NULL){
+
+        *head = new;
+        return;
+    }
+
+    for(; ptr->next != NULL; ptr = ptr->next);
+    
+    ptr->next = new;
+}
+
+/**
+ *@brief Appends new nodes with provided number element from randomized array
+ *@param[in] head Current Head of list 
+ *@param[in] arr Array of randomized numbers
+ *@param[in] size Array size 
+ *
+ */
+void populate_list(node_t **head, int *arr, int size){
+
+   for(int i = 0; i < size; i++)append_node(head, arr[i]); 
+
+}
+
+/**
+ *@brief This function inserts the new node in an alphabetical order
+ *@param[in] sorted Head of sorted list
+ *@param[in] new Current node element  
+ *
+ */
+void sorted_insert(node_t **sorted, node_t *new){
```

Comment:
Fügt ein Element in eine liste ein wobei die sortierungder Zahlen berücksichtigt werden